### PR TITLE
net: merge tcp, udp, uds features to net feature

### PR DESCRIPTION
**Pull request by taiki-e**
Refs: #2942

### CODE REVIEW

File: tokio/Cargo.toml [hunk 1]
```diff
@@ -54,7 +54,15 @@ io-util = ["memchr"]
...
 macros = ["tokio-macros"]
-net = ["dns", "tcp", "udp", "uds"]
+net = [
+  "dns",
+  "lazy_static",
+  "libc",
+  "mio/os-poll",
+  "mio/tcp",
+  "mio/udp",
+  "mio/uds",
```
Comment (LucioFranco) [inline tokio/Cargo.toml:64]: I feel like if I want tcp I really don't want to compile `mio-uds`?

Comment (taiki-e) [reply tokio/Cargo.toml:64]: It doesn't add [any new dependencies](https://github.com/tokio-rs/mio/blob/a5381b3e477ece2ef0db57adafad145746bd33cb/Cargo.toml#L29-L35). (also, we already [depends on `libc` via `mio`.](https://github.com/tokio-rs/mio/blob/a5381b3e477ece2ef0db57adafad145746bd33cb/Cargo.toml#L41)) I don't think they actually have a visible effect on compile time.

Comment (LucioFranco) [reply tokio/Cargo.toml:64]: That is fair, ill leave this up to @carllerche

**Review by Darksonn — APPROVED** [review]
I have no concerns besides the ones already listed.

File: tokio/Cargo.toml (hunk 1 above)
Comment (carllerche) [reply tokio/Cargo.toml:64]: I think it is fine given the size of `uds`. we can always split it up later if we need to again. We have no metrics on compile-time & bin size w/ various feature flags, so I would opt for reducing here.

**Review by carllerche — APPROVED** [review]
Looks great to me :+1: Let's merge this after #2876

**Pull request merged**